backpack/core/derivatives/rnn.py

```diff
@@ -0,0 +1,236 @@
+"""Partial derivatives for the torch.nn.RNN layer."""
+from typing import List, Tuple
+
+import torch
+from torch import Tensor, cat, einsum, zeros
+from torch.nn import RNN
+
+from backpack.core.derivatives.basederivatives import BaseParameterDerivatives
+
+
+class RNNDerivatives(BaseParameterDerivatives):
+    """Partial derivatives for the torch.nn.RNN layer.
+
+    a_t = W_ih x_t + b_ih + W_hh h_{t-1} + b_hh
+    h_t = tanh(a_t)
+
+    Index conventions:
+    ------------------
+    * t: Sequence dimension
+    * v: Free dimension
+    * n: Batch dimension
+    * h: Output dimension
+    * i: Input dimension
+    """
+
+    @staticmethod
+    def _check_parameters(module: RNN) -> None:
+        """Check the parameters of module.
+
+        Args:
+            module: module which to check
+
+        Raises:
+            ValueError: If any parameter of module does not match expectation
+        """
+        if module.num_layers > 1:
+            raise ValueError("only num_layers = 1 is supported")
+        if not module.nonlinearity == "tanh":
+            raise ValueError("only nonlinearity = tanh is supported")
+        if module.bias is not True:
+            raise ValueError("only bias = True is supported")
+        if module.batch_first is not False:
+            raise ValueError("only batch_first = False is supported")
+        if not module.dropout == 0:
+            raise ValueError("only dropout = 0 is supported")
+        if module.bidirectional is not False:
+            raise ValueError("only bidirectional = False is supported")
+
+    @staticmethod
+    def _a_jac_t_mat_prod(
+        output: Tensor,
+        weight_hh_l0: Tensor,
+        mat: Tensor,
+    ) -> Tensor:
+        """Calculates jacobian vector product wrt a.
+
+        Args:
+            output: the values of the hidden layer
+            weight_hh_l0: weight matrix hidden-to-hidden
+            mat: matrix to multiply
+
+        Returns:
+            jacobian vector product wrt a
+        """
+        V: int = mat.shape[0]
+        N: int = mat.shape[2]
+        T: int = mat.shape[1]
+        H: int = mat.shape[3]
+        a_jac_t_mat_prod: Tensor = zeros(V, T, N, H, device=mat.device)
+        for t in range(T)[::-1]:
+            if t == (T - 1):
+                a_jac_t_mat_prod[:, t, ...] = einsum(
+                    "vnh, nh -> vnh",
```

Comment:
The code so far consistently uses no white space in `einsum` equations. Can you adapt all equations accordingly and remove the white space?